fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBLuceneFileReference.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * FDBLuceneFileReference.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2021 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.lucene.directory;
+
+import com.apple.foundationdb.annotation.API;
+import com.apple.foundationdb.tuple.Tuple;
+
+import javax.annotation.Nonnull;
+
+/**
+ * A File Reference record laying out the id, size, and block size.
+ */
+@API(API.Status.EXPERIMENTAL)
+public class FDBLuceneFileReference {
+    private final long id;
+    private final long size;
+    private final long blockSize;
+
+    public FDBLuceneFileReference(@Nonnull Tuple tuple) {
+        this(tuple.getLong(0), tuple.getLong(1), tuple.getLong(2));
+    }
+
+    public FDBLuceneFileReference(long id, long size, long blockSize) {
+        this.id = id;
+        this.size = size;
+        this.blockSize = blockSize;
+    }
+
+    public long getId() {
+        return id;
+    }
+
+    public long getSize() {
+        return size;
+    }
+
+    public long getBlockSize() {
+        return blockSize;
+    }
+
+    public Tuple getTuple() {
```

Comment:
Discussed offline, we'll create a followup issue to change this to using a protobuf instead of a tuple for better future proofing, before it gets any real use.